### PR TITLE
User can now edit machines and sensors

### LINKED ISSUE #65: Update Machines and Sensor Frontend
As a User, I want to change some existing machines and sensors so that they better reflect the machines and sensors in real life

**Acceptance Criteria**

 - Users can change machine names and images in the front end
 - Users can change sensor name in the front end

**Is your feature request related to a problem? Please describe.**
Currently, users cannot change any names or images for either machines nor sensors. This means that once they create a machine/sensor, the name is stuck like that, without the ability to fix it or change it if an error occurs.

**Describe the solution you'd like**
Most likely some kind of modal, with an edit button for the machines/sensors. This modal should allow users to change the machine name and image, or the sensor name. Once the user has put in their inputs, the new name/image should be persisted onto the database.

**Describe alternatives you've considered**
N/A
**Additional context**
Need to look at the graphQL schemas for the mutations and see what kind of inputs they put. 
Sensor Update Input:
image

Machine Update Input: (with image)
image

### PR DESCRIPTION
**Pull request by raghavgarg32**
## GitHub Issue Solved:

closes #65 

## Current behaviour

User can't update machine names and images and sensor names

## Changed behaviour

User can now update machine names and images and sensor names

## PR checklist

Remember to check the following

 - [x] Tag specific people to review your PR
 - [ ] Update the ReadMe with any new run instructions
 - [x] Closes the associated issue (if applicable)
 - [ ] Updated kanban board

## Notes

image

image

### CODE REVIEW

**Review by KimberleyEvans-Parker — CHANGES_REQUESTED** [review]
Looking good 😄 I just have a couple of suggestions to improve the user experience.  
I noticed when the edit box opens, it doesn't contain a name.  This could lead to accidentally removing the name if you were to only add an image, but not change the name.  Is it possible to fill the edit name text with the current sensor name?
It would also be nice if you could add a fade on hover, or something similar to the edit buttons, so you know which area to click on.

**Comment by raghavgarg32** [comment]
> Looking good 😄 I just have a couple of suggestions to improve the user experience.
> I noticed when the edit box opens, it doesn't contain a name. This could lead to accidentally removing the name if you were to only add an image, but not change the name. Is it possible to fill the edit name text with the current sensor name?
> It would also be nice if you could add a fade on hover, or something similar to the edit buttons, so you know which area to click on.

Currently, if you only add an image and not a name, I made sure that the name isn't removed but only the image changes. Do you think I should still add the filled-in text for the name? I'll add a hover effect to the edit button.

File: app/src/components/MachineGrid.tsx [hunk 1]
```diff
@@ -24,13 +24,13 @@ const MachineGrid: React.FC<GridProps> = ({ allMachines, subscribedMachines, sho
...
-                                    name={machine?.name}
-                                    health={machine?.healthStatus}
-                                    image={machine?.image}
-                                />
-                            </Link>
+                            <MachineContainer
+                                key={machine.id}
+                                name={machine.name}
+                                health={machine.healthStatus}
+                                image={machine.image || "random"} // random is a placeholder right now, as not all machiens have images
```
Comment (FrankWangma) [inline app/src/components/MachineGrid.tsx:31]: I think we can remove the || "random"} now, since all machines have images (and will have images)

**Review by FrankWangma — CHANGES_REQUESTED** [review]
Everything's looking good. Just need to change the commented bit and i'll merge

File: app/src/components/MachineGrid.tsx (hunk 1 above)
Comment (raghavgarg32) [reply app/src/components/MachineGrid.tsx]: I've now removed it

**Review by FrankWangma — APPROVED** [review]
LGTM

**Pull request merged**